wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitJsonRpcMessenger.java

```diff
@@ -50,21 +53,34 @@ private void unsubscribe() {
   }
 
   @Override
-  public void onEvent(GitCheckoutEvent event) {
-    String workspaceIdAndProjectName = event.getWorkspaceId() + event.getProjectName();
-    endpointIds
-        .entrySet()
-        .stream()
-        .filter(it -> it.getValue().contains(workspaceIdAndProjectName))
-        .map(Entry::getKey)
-        .forEach(
-            it ->
-                transmitter
-                    .newRequest()
-                    .endpointId(it)
-                    .methodName("git/checkoutOutput")
-                    .paramsAsDto(event)
-                    .sendAndSkipResult());
+  public void onEvent(GitEvent event) {
+    if (event instanceof GitCheckoutEvent) {
```

Comment:
Make `onEvent` method simpler.  Encapsulate handling different git events in methods. 
```
if () {
  handleEvent1()
} else () {
  handlerEvent2()
}
```

Reply:
done